### PR TITLE
Image resizing and compression to webp

### LINKED ISSUE #51: Compress uploaded images
Profile images, event banners and club icons should be compressed when uploaded and potentially converted to jpg (extra compression), although that would remove transparency. Maybe we can convert all transparency to white when converting to jpg.

### PR DESCRIPTION
**Pull request by GRA0007**
Closes #51 

Uses the `sharp` package to resize and compress images when uploaded. Also crops images for users and clubs to a square. Also deletes old images from the bucket.

Does image type checking on the frontend as well.

### CODE REVIEW

**Review by Prydeton — APPROVED** [review]
LGTM

File: backend/src/util/uploadFile.js [hunk 1]
```diff
@@ -11,64 +13,138 @@ const storage = new Storage({
 
 const bucket = storage.bucket(bucketName)
 
+const isImage = mimetype => [
```
Comment (giraugh) [inline backend/src/util/uploadFile.js:16]: As per Discord, this is somewhat unreliable from a security perspective. For this use case however its fine as `sharp` will throw errors for malformed images. We just want to keep this in mind for future non-image file uploading.

File: backend/src/util/uploadFile.js [hunk 2]
```diff
@@ -11,64 +13,138 @@ const storage = new Storage({
...
+  'image/svg+xml',
+  'image/webp',
+  'image/tiff',
+  'image/gif',
+].includes(mimetype)
+
+const humanFileSize = size => {
+  const i = size === 0 ? 0 : Math.floor(Math.log(size) / Math.log(1024))
+  return (size / Math.pow(1024, i)).toFixed(2) * 1 + ' ' + ['B', 'kB', 'MB', 'GB', 'TB'][i]
+}
```
Comment (giraugh) [inline backend/src/util/uploadFile.js:29]: This is a very nice touch :ok_hand:

**Review by giraugh — APPROVED** [review]
Very nice 😎 
LGTM 👍

File: backend/src/util/uploadFile.js (hunk 1 above)
Comment (GRA0007) [reply backend/src/util/uploadFile.js:16]: We can track this in #321

**Pull request merged**